### PR TITLE
Better support comments in mysql files

### PR DESCRIPTION
**Pull request by mchugh19**
### What does this PR do?
Better handles comments in mysql files. 
Also adds a no_parse flag to pass the entirety of the mysql file contents to mysql. This allows mysql to handle its own parsing, which might work better for some complex setups, but also prevents salt from returning specifics of what has changed.

### What issues does this PR fix or reference?
#47488

### Previous Behavior
A file like
```
-- some comment
INSERT INTO users (username) VALUES ("sampleuser2");  # after
#another comment
INSERT INTO users (username) VALUES ("sampleuser3"); 
```
would be interpreted as
```
INSERT INTO users (username) VALUES ("sampleuser2");
# after #another comment INSERT INTO users (username) VALUES ("sampleuser3");
```

### New Behavior
Now that file is interpreted as
```
INSERT INTO users (username) VALUES ("sampleuser2");
INSERT INTO users (username) VALUES ("sampleuser3");
```

When no_parse is used, the execution module will return something like the following for files which contain many create, insert, and update statements.
```
local:
    ----------
    query time:
        ----------
        human:
            0.0s
        raw:
            0.00042
    rows affected:
        1  
```
### Tests written?
Yes - existing test updated to support this functionality

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by mchugh19** [comment]
Rebased on master

**Comment by DmitryKuzmenko** [comment]
@mchugh19 thank you for this work! It's awesome!
It's not a full support of MySQL/MariaDB comments style, but it's much better than it was.
The only thing I want to ask you. You're using regex that is as powerful as hardly readable and supportable. Meanwhile our tests don't cover your functionality. You've added a generic regression update to an existing integration test. But we definitely need tests:
1. checking your regular expressions in all combinations of data;
2. covering the `no_parse` case.

There's already a unit test for mysql module so it'd be pretty simple to extend this.
Are you able to extend the test to your functionality please?

**Comment by DmitryKuzmenko** [comment]
Additional thanks for blacken your code!

**Comment by mchugh19** [comment]
Never say nothing good came out of the test cases! Since getting back to this, I was not able to get the `no_parse` feature to work with these libraries. Since it's that flakey, I just removed the whole thing. An additional unit test was written for the comment handling.

File: tests/unit/modules/test_mysql.py [hunk 1]
```diff
@@ -620,6 +620,29 @@ def test_plugin_status(self):
...
+        Test comment sanitization
+        """
+        input_data = """/*
+        multiline
+        comment
+        */
+        CREATE TABLE test_update (a INT); # end of line comment
+        # example comment
+        insert into test_update values (1); -- ending comment
+        -- another comment type
```
Comment (DmitryKuzmenko) [inline tests/unit/modules/test_mysql.py:634]: Just one more case please-please. A comment sign inside the quotes.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
Please understand me correctly I'm not trying to bore you. =) Thanks for understanding!

File: tests/unit/modules/test_mysql.py (hunk 1 above)
Comment (mchugh19) [reply tests/unit/modules/test_mysql.py:635]: Is this what you meant? (see commit)

Comment (DmitryKuzmenko) [reply tests/unit/modules/test_mysql.py:635]: Perfect! Thank you!

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by DmitryKuzmenko** [comment]
This PR is blacken and merge ready.

**Pull request merged**